test/PaymasterCommitment.test.ts

```diff
@@ -189,6 +189,53 @@ contract('Paymaster Commitment', function ([_, relayOwner, relayManager, relayWo
       assert.closeTo(paid, gasUsed, 50)
     })
 
+    it('paymaster should not pay for requests exceeding msg.data size limit', async () => {
+      const r = await makeRequest(web3, {
+        request: {
+          // nonce: '4',
+          data: recipientContract.contract.methods.emitMessage('').encodeABI()
+        },
+        relayData: { paymaster }
+
+      }, sharedRelayRequestData, chainId, forwarderInstance)
+      const gasAndDataLimits = await paymasterContract.getGasAndDataLimits()
+      const hugeApprovalData = '0x' + 'ef'.repeat(parseInt(gasAndDataLimits.calldataSizeLimit))
+      await expectRevert(
+        relayHubInstance.relayCall(10e6, r.req, r.sig, hugeApprovalData, externalGasLimit, {
+          from: relayWorker,
+          gas: externalGasLimit,
+          gasPrice
+        }), 'msg.data exceeded limit'
+      )
+    })
+
+    it('paymaster should not pay for requests with max msg.data size if it rejects in pre', async () => {
```

Comment:
I do not see a corresponding change in the Relay Hub, indicating this test is not correct.

Reply:
See line 136 in RelayHub.sol. We added max calldata size, hub as this `require`. This is literally the edge case You asked me to add, when wee send the maximum size and paymaster rejects.

Reply:
The description is incorrect then. This test "paymaster should not pay for requests with max msg.data size if it rejects in pre" is hard or impossible to implement. I cannot understand what is actually checked here.

Reply:
You're the one who specifically asked me to add this test for the edge case where the msg.data.length reaches limit and paymaster rejects. Do you want me to remove it?

Reply:
Oh, ok. I understand now. Test description was not clear, or maybe I was drunk, I don't know.